### ISSUE TITLE
Use --allow-empty, remove echo from generated gitfiti.sh

### ISSUE THREAD

**Issue by numberoverzero** [gelstudios/gitfiti#32]
## tl;dr

Stop generating `echo` commands and use `--allow-empty` instead of `-a` when generating commit commands.

For a run that exceeds an existing max of 38, this saves ~50% runtime (1m46s -> 53s) and ~99% space (213M -> 1.7M) before compression.  Reducing the commits to 0 changes makes it easier for git to apply each commit and compress the commits for upload to github, which account for most of the current runtime.

## change

After comparing the output (and runtime) of the generated gitfiti.sh and the script from [Pokrovsky's Github History Vandalism as a service](http://pokrovsky.herokuapp.com/) I noticed that the latter uses the [`--allow-empty`](https://github.com/pikesley/pokrovsky/blob/ea4e0b9a8bc34740046854192f4a0238a4f13842/lib/pokrovsky/commit.rb#L7) option, which removes the need for `echo [integer] >> gitfiti` entirely.

Simply change these:

```
echo 0 >> gitfiti
GIT_AUTHOR_DATE=... GIT_COMMITTER_DATE=... git commit -a -m "gitfiti" > /dev/null
```

to this:

```
GIT_AUTHOR_DATE=... GIT_COMMITTER_DATE=... git commit --allow-empty -m "gitfiti" > /dev/null
```

## Data

Time with the existing generator:

```
$ time bash gitfiti.sh
Initialized empty Git repository in $REPO/.git/
Auto packing the repository for optimum performance. You may also
run "git gc" manually. See "git help gc" for more information.
Counting objects: 51601, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (51581/51581), done.
Writing objects: 100% (51601/51601), done.
Total 51601 (delta 17173), reused 0 (delta 0)
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> master

Counting objects: 51601, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (34408/34408), done.
Writing objects: 100% (51601/51601), 4.12 MiB | 2.17 MiB/s, done.
Total 51601 (delta 17173), reused 51601 (delta 17173)
To git@github.com:numberoverzero/$REPO.git
 * [new branch]      master -> master
Branch master set up to track remote branch master from origin.

real	1m46.578s
user	0m43.800s
sys	1m2.294s
```

Time without `echo`, using `--allow-empty`:

```
$ time bash gitfiti.sh
Initialized empty Git repository in $REPO/.git/
Auto packing the repository for optimum performance. You may also
run "git gc" manually. See "git help gc" for more information.
Counting objects: 17202, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (17201/17201), done.
Writing objects: 100% (17202/17202), done.
Total 17202 (delta 17187), reused 0 (delta 0)
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> master

Counting objects: 17202, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (14/14), done.
Writing objects: 100% (17202/17202), 1.02 MiB | 0 bytes/s, done.
Total 17202 (delta 17187), reused 17202 (delta 17187)
To git@github.com:numberoverzero/$REPO.git
 * [new branch]      master -> master
Branch master set up to track remote branch master from origin.

real	0m53.973s
user	0m7.907s
sys	0m44.794s
```

Size of generated .git with existing generator:

```
$ du -sch $REPO/.git
213M    $REPO/.git
213M    total
```

Size of generated .git without `echo`, using `--allow-empty`:

```
du -sch $REPO/.git
1.7M    $REPO/.git
1.7M    total
```

**Comment by gelstudios** [comment]
Awesome! I didn't realize git allowed empty commits, do they show up in the
commit calendar?

**Comment by numberoverzero** [comment]
Re-ran to verify - my graph reverted, then I ran the modified gitfiti.sh and it's back.  Here's the generated repo: https://github.com/numberoverzero/github-history-vandalism

**Comment by gelstudios** [comment]
It does concern me a little bit that the commit history for that file in your repo says: "Showing 0 changed files with 0 additions and 0 deletions".

But if it counts for the calendar, sounds good to me. 👍

**Comment by gelstudios** [comment]
Do you want to try your hand at a PR for this?

**Comment by numberoverzero** [comment]
Sorry for the slow response - I'll give it a shot.

**Comment by numberoverzero** [comment]
Alright, looks good.  I ran the new generated script against [numberoverzero/foo-dev](https://github.com/numberoverzero/foo-dev) and it overlaid the same oneup2 as expected (offset by about a week, since I last ran it).

I'll delete that repo once the PR is merged.

**Comment by numberoverzero** [comment]
PR is up: https://github.com/gelstudios/gitfiti/pull/42
